chain/chain/src/store.rs

```diff
@@ -706,6 +706,45 @@ impl ChainStore {
         //    2.2. Parse the trie key with a relevant KeyFor* implementation to ensure consistency
 
         Ok(match state_changes_request {
+            StateChangesRequest::AllAccountsChanges { block_header } => {
+                let state_changes: Vec<StateChangeKindView> = self
+                    .get_state_changes_in_block(&block_header.hash)?
+                    .into_iter()
+                    .map(Into::into)
+                    .collect();
+
+                // TODO(mina86): Do we actually need ‘seen’?  I’m kinda confused at this
+                // point how changes are stored in the database and whether view_client can
+                // return duplicate AccountTouched entries.
+                let mut seen = std::collections::HashSet::new();
+                let touched_account_ids = state_changes
+                    .into_iter()
+                    .filter_map(|x| {
+                        if let near_primitives::views::StateChangeKindView::AccountTouched {
+                            account_id,
+                        } = x
+                        {
+                            Some(account_id)
+                        } else {
+                            None
+                        }
+                    })
+                    .filter(move |account_id| {
+                        // TODO(mina86): Convert this to seen.get_or_insert_with(account_id,
+                        // Clone::clone) once hash_set_entry stabilises.
+                        seen.insert(account_id.clone())
+                    })
+                    .collect::<Vec<_>>();
+
+                let all_accounts_changes = self
+                    .get_state_changes(
+                        &block_header.hash,
+                        &StateChangesRequest::AccountChanges { account_ids: touched_account_ids },
+                    )
+                    .expect("All accounts state changes received");
+
+                all_accounts_changes
+            }
```

Comment:
Moving the code verbatim doesn’t really buy us much though. This needs to be reimplemented such that it does only one iteration over the trie. This can be done by adding `TrieKey::AllAccounts` variant which would correspond to prefix `block_hash + col::ACCOUNT`. (Currently we have `TrieKey::Account` which corresponds to `black_hash + col::ACCOUNT + account_id`). Then, `AllAccountsChanges` would be handled by something along the lines of:


```rust
let data_key = TrieKey::AllAccounts;
let storage_key = KeyForStateChanges::from_trie_key(block_hash, &data_key);
let changes_per_key_prefix = storage_key.find_iter(&self.store);
let changes = StateChanges::from_account_changes(changes_per_key_prefix)?
```

I think this should just work.